### ISSUE TITLE
How to get simple inference result as txt

### ISSUE THREAD

**Issue by halzhen** [ultralytics/yolov5#6382]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I am trying to inference on a batch of images in a directory. The basic inference code is the one used , I have already read the documentation for the Detections return, is there any way to get the detected objects as .txt ? In a similar way to result.print() , it shows the objects and the times they appear.

I am aware that detect.py can save as .txt but couldn't do inference on batch of images with it.

```# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')

# Images
directory = 'keyframes/'

frames = [] # batch of images
for filename in os.listdir(directory):
    frame = Image.open('keyframes/'+filename)
    frames.append(frame)

# Inference
results = model(frames, size=640) 

# Results
results.print()
results.save()  ```

What I need is the objects detected in a .txt file without using detect.py, but if detect.py can be modified to work on batch of images from a directory that would also work. 


### Additional

_No response_

**Comment by glenn-jocher** [comment]
@kzhenzhou that's funny, you are right there is no `--save-txt` method for PyTorch Hub models. You could submit a PR to add such a method, i.e.:

```
results.save_txt()
```

https://github.com/ultralytics/yolov5/blob/bd815d48df18a23e2bb08d88e430183bfb48eb78/models/common.py#L551-L650

I'll add a TODO here since this seems like a useful feature to have, but we are quite busy so I can not provide an estimated implementation timeline if we do this ourselves.

TODO: `results.save_txt()` method.

**Comment by SarperYurttas** [comment]
@glenn-jocher I want to work on this if it's ok?

**Comment by glenn-jocher** [comment]
@SarperYurttas awesome, yes go for it!

**Comment by Symbadian** [comment]
> @kzhenzhou that's funny, you are right there is no `--save-txt` method for PyTorch Hub models. You could submit a PR to add such a method, i.e.:
> 
> ```
> results.save_txt()
> ```
> 
> https://github.com/ultralytics/yolov5/blob/bd815d48df18a23e2bb08d88e430183bfb48eb78/models/common.py#L551-L650
> 
> I'll add a TODO here since this seems like a useful feature to have, but we are quite busy so I can not provide an estimated implementation timeline if we do this ourselves.
> 
> TODO: `results.save_txt()` method.

Hi @glenn-jocher, 

is there anyway to get the heading of the result type saved as well in the output files?

for example: 

epoch | loss | acc | val_loss | val_acc
            |        |         |               | 

Currently, in the save files the data is presented as the example below without headings in .txt format
0  1.64137483  0.27450982  0.8043378  0.46153846

I tried --save-txt with save --save-conf but none is presenting the headings for the processing results...

I am trying to do a basic decision process but  these bits are not available!

Thanx for responding and acknowledging my digital presence, really appreciate this!

**Comment by glenn-jocher** [comment]
@Symbadian Thanks for reaching out! We appreciate your support and contribution to the YOLOv5 community. Currently, the `--save-txt` option does not include column headings in the saved .txt file. However, you can easily manipulate the .txt file after saving it to include headings using any text manipulation tool or a simple script. We appreciate your understanding and contributions. Good luck with your work!
